qiskit/circuit/instruction.py

```diff
@@ -136,34 +137,36 @@ def params(self, parameters):
         self._params = []
         for single_param in parameters:
             # example: u2(pi/2, sin(pi/4))
-            if isinstance(single_param, (ParameterExpression, sympy.Basic)):
+            if isinstance(single_param, (ParameterExpression)):
                 self._params.append(single_param)
             # example: OpenQASM parsed instruction
             elif isinstance(single_param, node.Node):
-                self._params.append(single_param.sym())
+                self._params.append(single_param.num())
```

Comment:
Out of curiosity, do you know when we hit this code? It seems strange that we accept qasm nodes here. Maybe @1ucian0 knows?

Reply:
:+1: Seems reasonable, thanks for looking into it. A deprecation warning seems like the safer choice. Also, where is `.num` defined? I took a look through `qasm/`  and didn't see it referenced anywhere.

Reply:
Umm, you know what I don't think it's real. I can't find it either. I have no idea why I changed this or what I was thinking. I'll have to take a look through the dag node definitions to figure out what I was intending here. From a quick scan it looks like this code wasn't great either, the `sym()` method is only defined for a `binaryop`, `external`, `prefix`, `id`, `intnode`, and `real` node types which is a pretty small subset.

Reply:
Lol, actually I remember why it changed. I did a sloppy regex `s/sym/num/g` to replace `sympy` with `numpy`. I'll swithc it back. Looking through the definitions of `sym()` methods I think it should just work, and clearly it's not being used anywhere. I'll push up a separate pr to deprecate it.

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/3156/commits/3c1d2906703af2a55681a36862f0a6708d7e757d